Entities/Common/Controls/StandardControlsCommon.as

```diff
@@ -40,3 +40,61 @@ bool isTapPickup(CBlob@ this, int ticks = 15)
 	// TODO: merge some code with the above and make it generalized to all keys if ever useful
 	return (getGameTime() - this.get_s32("tap_pickup_time") < ticks);
 }
+
+
+bool PointInsidePolygon( Vec2f Point, Vec2f[] polyVerts, Vec2f polyPos, bool facingLeft )
+{
+	// Mirror the polygon when the blob is facing left
+	// We make a copy, as we do not want to modify the original
+    Vec2f[] polygon;
+
+    if (facingLeft) 
+    {
+      Vec2f[] mirrored;
+      for ( int i = 0 ; i < polyVerts.length ; i++ )
+      {
+        Vec2f q = Vec2f(-polyVerts[i].x, polyVerts[i].y);
+        mirrored.push_back(q);
+      }
+
+      polygon = mirrored;
+    } 
+    else 
+    {
+      polygon = polyVerts;
+    }
+
+    double minX = polyPos.x+polygon[0].x;
+    double maxX = polyPos.x+polygon[0].x;
+    double minY = polyPos.y+polygon[0].y;
+    double maxY = polyPos.y+polygon[0].y;
+
+    for ( int i = 1 ; i < polygon.length ; i++ )
+    {
+        Vec2f q = polyPos+polygon[ i ];
+        minX = Maths::Min( q.x, minX );
+        maxX = Maths::Max( q.x, maxX );
+        minY = Maths::Min( q.y, minY );
+        maxY = Maths::Max( q.y, maxY );
+    }
+
+    if ( Point.x < minX || Point.x > maxX || Point.y < minY || Point.y > maxY )
+    {
+        return false;
+    }
+
+    bool inside = false;
+    for ( int i = 0, j = polygon.length - 1 ; i < polygon.length ; j = i++ )
+    {
+    	Vec2f pvi = polyPos + polygon[ i ];
+    	Vec2f pvj = polyPos + polygon[ j ];
+        if ( ( pvi.y > Point.y ) != ( pvj.y > Point.y ) &&
```

Comment:
I'll trust that the math is correct as it appears to work :P

Reply:
I got that from somewhere on the ~~ internet ~~